lib/palindrome_check.rb

```diff
@@ -1,5 +1,20 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
 def palindrome_check(my_phrase)
-  raise NotImplementedError
-end
+  return false if my_phrase.nil?
+
+    j = my_phrase.length - 1
+    i = 0
+
+    while i < j
+     i += 1 while my_phrase[i] == " "
+
+    j -= 1  while my_phrase[j] == " "
```

Comment:
Continue to check that `i < j` before comparing the character at index  j with a white space in the nested loop.